src/Microsoft.NetCore.Analyzers/Core/Performance/UsePropertyInsteadOfCountMethodWhenAvailable.Fixer.cs

```diff
@@ -0,0 +1,115 @@
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+
+{
+    /// <summary>
+    /// CA1829: Use property instead of <see cref="Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})"/>, when available.
+    /// Implements the <see cref="Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer" />
+    /// </summary>
+    public abstract class UsePropertyInsteadOfCountMethodWhenAvailableFixer : CodeFixProvider
+    {
+        /// <summary>
+        /// A list of diagnostic IDs that this provider can provider fixes for.
+        /// </summary>
+        /// <value>The fixable diagnostic ids.</value>
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.RuleId);
+
+
+        /// <summary>
+        /// Gets an optional <see cref="FixAllProvider" /> that can fix all/multiple occurrences of diagnostics fixed by this code fix provider.
+        /// Return null if the provider doesn't support fix all/multiple occurrences.
+        /// Otherwise, you can return any of the well known fix all providers from <see cref="WellKnownFixAllProviders" /> or implement your own fix all provider.
+        /// </summary>
+        /// <returns>FixAllProvider.</returns>
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <summary>
+        /// Computes one or more fixes for the specified <see cref="CodeFixContext" />.
+        /// </summary>
+        /// <param name="context">A <see cref="CodeFixContext" /> containing context information about the diagnostics to fix.
+        /// The context must only contain diagnostics with a <see cref="Diagnostic.Id" /> included in the <see cref="CodeFixProvider.FixableDiagnosticIds" /> 
+        /// for the current provider.</param>
+        /// <returns>A <see cref="Task" /> that represents the asynchronous operation.</returns>
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            if (node is object &&
+                context.Diagnostics[0].Properties.TryGetValue(UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.PropertyNameKey, out var propertyName) &&
```

Comment:
`Properties` can be null, so you can go a step further and use `Diagnostics[0]?.Properties`

Reply:
As per our chat, `.Properties` is never null, so this is OK!